### PR TITLE
fix: [CDS-45475]: partially hidden error message in create pipeline modal / Edit Stage Modal

### PR DESCRIPTION
**Pull request by vikrant-harness**
### Summary

- To keep UX consistent the margin was removed but when error message pops-up it gets hidden behind the name-id field. so conditionally adding the margin when `formik` error populates
- This was fixed for CreatePipelineModal and EditStageModal 

#### Screenshots

https://user-images.githubusercontent.com/106532291/197769692-77c1e07a-2385-4c30-a61d-1d1f34b6c800.mov

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/70-pipeline/components/PipelineStudio/CreateModal/PipelineCreate.tsx [hunk 1]
```diff
@@ -173,7 +174,13 @@ export default function CreatePipelines({
                 isIdentifierEditable: pipelineIdentifier === DefaultNewPipelineId
               }}
               tooltipProps={{ dataTooltipId: 'pipelineCreate' }}
-              inputGroupProps={{ className: css.zeroMargin }}
+              inputGroupProps={{
+                ...(!(
+                  errorCheck('name', formikProps) || get(formikProps, `errors.${getString('identifier').toLowerCase()}`)
```
Comment (rameshx) [inline src/modules/70-pipeline/components/PipelineStudio/CreateModal/PipelineCreate.tsx:179]: Field names shouldn't be translated. Just use `errors.identifier`.

File: src/modules/75-cd/components/PipelineStudio/DeployStage/EditStageView/EditStageView.tsx [hunk 2]
```diff
@@ -347,7 +348,15 @@ export const EditStageView: React.FC<EditStageViewProps> = ({
                               identifierProps={{
                                 inputLabel: getString('stageNameLabel'),
                                 isIdentifierEditable: !context,
-                                inputGroupProps: { disabled: isReadonly, className: css.zeroMargin }
+                                inputGroupProps: {
+                                  disabled: isReadonly,
+                                  ...(!(
+                                    errorCheck('name', formikProps) ||
+                                    get(formikProps, `errors.${getString('identifier').toLowerCase()}`)
```
Comment (rameshx) [inline src/modules/75-cd/components/PipelineStudio/DeployStage/EditStageView/EditStageView.tsx:355]: this can be moved to a const as it is repeated

**Review by rameshx — CHANGES_REQUESTED** [review]

File: src/modules/75-cd/components/PipelineStudio/DeployStage/EditStageView/EditStageView.tsx (hunk 2 above)
Comment (vikrant-harness) [reply src/modules/75-cd/components/PipelineStudio/DeployStage/EditStageView/EditStageView.tsx:355]: done

File: src/modules/70-pipeline/components/PipelineStudio/CreateModal/PipelineCreate.tsx (hunk 1 above)
Comment (rameshx) [reply src/modules/70-pipeline/components/PipelineStudio/CreateModal/PipelineCreate.tsx]: @vikrant-harness Formik field names are part of the code and end up as name attributes on the input elements. We should only translate strings that the end user will see, hear, etc.

**Review by rameshx — DISMISSED** [review]

**Review by YounixIO — DISMISSED** [review]

**Review by YounixIO — APPROVED** [review]

**Review by rameshx — APPROVED** [review]

**Review by PratyushGar — APPROVED** [review]

**Review by Tamaraius — APPROVED** [review]

**Pull request merged**